src/Features/LanguageServer/Protocol/Handler/Definitions/GoToDefinitionHandlerBase.cs

```diff
@@ -47,15 +52,15 @@ internal static class GoToDefinitionHandler
                     });
                 }
             }
-            else if (document.SupportsSemanticModel && metadataAsSourceService != null)
+            else if (document.SupportsSemanticModel && _metadataAsSourceFileService != null)
```

Comment:
so, if we use SymbolNavigationService, we could do the decompliation stuff as well. based on option.. not sure whether we will have client option different than host option or client always follow host option. and how that works with other scenario. work item?

Reply:
https://github.com/dotnet/roslyn/projects/45#card-20873103